lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -440,4 +443,32 @@
 
         return undefined;
     });
+
+    platform.registerFunction(FunctionsEnum.ObjectFromEntries, function (iterable) {
+        // #sec-object.fromentries
+        "use strict";
+        if (iterable === null || iterable === undefined) {
+            __chakraLibrary.raiseNeedObject("Object.fromEntries");
+        }
+
+        const o = {};
+        const propDescriptor = {
+            enumerable : true,
+            configurable : true,
+            writable : true,
+            value : undefined
+        };
+
+        let key;
+        for (const entry of iterable) {
+            if (typeof entry !== "object") {
```

Comment:
Hm. You may want to add `|| entry === null` to this check, just so you get a better error message. It's the right semantics either way - if entry is null, it will throw a TypeError at the `key = entry[0]` line - but it may be surprising that the messages differ when entry is null vs any other primitive, especially since those are forbidden by the same part of the spec.

Reply:
Currently the error message for failing on the key - entry[0] line is better than the error message for failing this check, so would rather not update unless devising a new specific error.

Note CC will report that the error occurs in the line of user code containing the fromEntries call - the code in JsBuiltIn.js is hidden from the end user of CC.

The raiseNeedObject error simply gives "Object expected" as the error message whereas key = entry[0] line will give the slightly more helpful "Unable to get property '0' of undefined or null reference".

If combining them into one error it should probably be something specific like "Iterable provided to Object.fromEntries must not return non-object or null value"

Reply:
For what its worth, it is possible to add a new error :)